### PR TITLE
Improved rendering of combinations in Product page

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.3.x
| Description?  | In product page, the retrieval and rendering of Combinations are so slow that the page is unusable
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | yes `CombinationDataProvider::getFormCombination`
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-2950
| How to test?  | Create more than 150 combinations with the combination generator, then try to display and update combinations: you should notice a performance boost. ~~Also, if the number of combination is above 50, we set the `max_input_vars value` dynamically~~.

error_message

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
* Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8871)

### CODE REVIEW

File: src/Adapter/CombinationDataProvider.php [hunk 1]
```diff
@@ -51,6 +51,7 @@ public function __construct()
 
     /**
      * Get a combination values
+     * @deprecated since 1.7.4 really slow, use getFormCombinations instead.
```
Comment (eternoendless) [inline src/Adapter/CombinationDataProvider.php]: To be changed to 1.7.3.1

**Review by eternoendless — DISMISSED** [review]

**Review by eternoendless — DISMISSED** [review]

**Comment by marionf** [comment]
@mickaelandrieu 

I generated 168 combinations, the combination list is well displayed but when I try to edit the quantity for a combination I still have the invalid CSRF token error.

capture du 2018-03-26 11-41-54

If I leave the page and return on it again, the combinations are not displayed anymore

capture du 2018-03-26 11-43-33

**Comment by mickaelandrieu** [comment]
Hello @marionf,

> I generated 168 combinations, the combination list is well displayed but when I try to edit the quantity for a combination I still have the invalid CSRF token error.

Yes, you need to increase the `max_input_vars` from 1000 to 15000 in your PHP configuration, this is a limitation of PHP which - by default - can manage only 1000 fields.

So when you have more than 1000 fields in the page, PHP ignore all the additional fields, including the csrf ones: this is why you see this error. I wanted to fix this bug too, but you can't do it in PrestaShop code as this value can only be changed in `php.ini` file.

ping @eternoendless: as we can evaluate the number of fields displayed in Product Page, shouldn't we compare the value with `max_input_vars` and display a meaningful error, at least for developers?

**Comment by eternoendless** [comment]
> shouldn't we compare the value with max_input_vars and display a meaningful error, at least for developers?

Good idea!

**Comment by marionf** [comment]
@mickaelandrieu 

Ok, I don't have the CSRF token error if I increase the max_input_vars.
But I don't have the list of combinations if I leave the page and return (even with only 44 combinations)
No errors in the console.

capture du 2018-03-26 15-47-40

**Comment by mickaelandrieu** [comment]
@marionf, you're right, I needed to update the pull request as this one was intended for `develop` branch ... sorry for that, can you re-check, please?

File: src/PrestaShopBundle/Controller/Admin/ProductController.php [hunk 2]
```diff
@@ -442,7 +442,22 @@ public function formAction($id, Request $request)
...
+
         if (is_array($combinations)) {
+            $maxInputVars = (int) ini_get('max_input_vars');
+            $combinationsCount = count($combinations) * 25;
+            $combinationsInputs = ceil($combinationsCount/1000)*1000;
+
+            if ($combinationsInputs > $maxInputVars) {
+
+                $this->addFlash('error', $this->trans(
+                    'The value of "max_input_vars" should be increased to %value% in order to be able to submit the product form.',
```
Comment (eternoendless) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php]: How about this:

> The value of the PHP.ini setting "max_input_vars" must be increased to %value% in order to be able to submit the product form.

Comment (mickaelandrieu) [reply src/PrestaShopBundle/Controller/Admin/ProductController.php]: done!

**Review by eternoendless — APPROVED** [review]

**Comment by eternoendless** [comment]
Thank you @mickaelandrieu

**Pull request merged**